src/sentry/preprod/api/endpoints/project_preprod_build_details.py

```diff
@@ -61,6 +66,41 @@ def get(self, request: Request, project, artifact_id) -> Response:
         except PreprodArtifact.DoesNotExist:
             return Response({"error": f"Preprod artifact {artifact_id} not found"}, status=404)
 
+        try:
+            size_metrics_qs = PreprodArtifactSizeMetrics.objects.select_related(
+                "preprod_artifact"
+            ).filter(
+                preprod_artifact__project=project,
+                preprod_artifact__id=artifact_id,
+            )
+            size_metrics_count = size_metrics_qs.count()
+            if size_metrics_count == 0:
+                logger.info("No size analysis results found for preprod artifact %s", artifact_id)
+                size_info = None
+            elif size_metrics_count > 1:
+                logger.info(
+                    "Multiple size analysis results found for preprod artifact %s", artifact_id
+                )
+                size_info = None
+            else:
+                size_metrics = size_metrics_qs.first()
+                if size_metrics.min_install_size is None or size_metrics.min_download_size is None:
+                    logger.info(
+                        "Size analysis results found for preprod artifact %s but no min install or download size",
+                        artifact_id,
+                    )
+                    size_info = None
+                else:
+                    size_info = BuildDetailsSizeInfo(
+                        install_size_bytes=size_metrics.min_install_size,
+                        download_size_bytes=size_metrics.min_download_size,
+                    )
+        except Exception:
+            return Response(
```

Comment:
Does this automatically propagate to a logger somewhere?

Reply:
we can validate this once we are finally in production, but I'd imagine the Sentry logs tool would include response bodies similar to how our old datadog <> serverless setup did